packages/smooth_app/lib/generic_lib/widgets/smooth_error_card.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_app/generic_lib/buttons/smooth_simple_button.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_card.dart';
+import 'package:smooth_app/helpers/extension_on_text_helper.dart';
+
+class SmoothErrorCard extends StatefulWidget {
+  const SmoothErrorCard({Key? key, required this.errorMessage, required this.tryAgainFunction})
+      : super(key: key);
+  final String errorMessage;
+  final void Function() tryAgainFunction;
+
+  @override
+  State<SmoothErrorCard> createState() => _SmoothErrorCardState();
+}
+
+class _SmoothErrorCardState extends State<SmoothErrorCard> {
+  late AppLocalizations _appLocalizations;
+  final double _buttonsHeight = 40;
+  final double _buttonsWidth = 200;
+  final TextStyle _buttonsTextStyle = const TextStyle(color: Colors.white);
+  bool _showErrorText = false;
+
+  void _setShowErrorText() {
+    setState(() {
+      _showErrorText = !_showErrorText;
+    });
+  }
+
+  Widget _getTryAgainButton() {
+    return SmoothSimpleButton(
+      height: _buttonsHeight,
+      child: Text(
+        _appLocalizations.try_again,
+        style: _buttonsTextStyle,
+      ),
+      onPressed: widget.tryAgainFunction,
+      minWidth: _buttonsWidth,
+    );
+  }
+
+  Widget _getErrorMessage() {
+    return Container(
+      margin: const EdgeInsets.all(10),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          const Icon(
+            Icons.error_outline,
+            size: 40,
+            color: Colors.red,
+          ),
+          Expanded(
+            child: Padding(
+              padding:
+                  const EdgeInsets.symmetric(horizontal: 8.0, vertical: 4.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  Text(
+                    _appLocalizations.error_occurred,
+                    style: const TextStyle(
+                      color: Colors.red,
+                      fontWeight: FontWeight.w500,
+                    ),
+                  ),
+                  const SizedBox(height: 20),
+                  Text(
+                    widget.errorMessage,
+                    style: const TextStyle(
+                      fontWeight: FontWeight.w500,
+                    ),
```

Comment:
I fear that this will be hard to read in darkmode. Something like `Theme.of(context).textTheme.bodyText2.apply(fontWeight: FontWeight.w500,),` would probably do the trick